src/pt/mangastop/src/eu/kanade/tachiyomi/extension/pt/mangastop/MangaStop.kt

```diff
@@ -23,16 +24,21 @@ class MangaStop : MangaThemesia(
         .rateLimit(3)
         .build()
 
-    override fun pageListParse(document: Document): List<Page> {
-        val pages = super.pageListParse(document)
-            .filterNot { it.imageUrl?.contains("mihon", true) == true }
+    override fun pageListParse(response: Response): List<Page> {
+        val html = response.body.string()
 
-        if (pages.isNotEmpty()) return pages
+        MangaThemesia.JSON_IMAGE_LIST_REGEX.find(html)?.groupValues?.get(1)?.let {
+            runCatching {
+                val pages = json.parseToJsonElement(it).jsonArray.mapIndexed { i, jsonEl ->
+                    Page(i, response.request.url.toString(), jsonEl.jsonPrimitive.content)
+                }
+                if (pages.isNotEmpty()) return pages
+            }
+        }
 
-        return MangaThemesia.JSON_IMAGE_LIST_REGEX.find(document.toString())
-            ?.groupValues?.get(1)
-            ?.let { json.parseToJsonElement(it).jsonArray }
-            ?.mapIndexed { i, el -> Page(i, document.location(), el.jsonPrimitive.content) }
-            .orEmpty()
+        val document = Jsoup.parse(html, response.request.url.toString())
+        return super.pageListParse(document).filterNot {
```

Comment:
If you're going to call `super.pageListParse(document: Document)` (instead of `super.pageListParse(response: Response)`), I would just use the previous version.